pkg/receiver/mysqlrecordsreceiver/client.go

```diff
@@ -0,0 +1,343 @@
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+package mysqlrecordsreceiver
+
+import (
+	"context"
+	"crypto/tls"
+	"crypto/x509"
+	"database/sql"
+	"encoding/json"
+	"io/ioutil"
+	"time"
+
+	"strconv"
+	"strings"
+
+	"github.com/aws/aws-sdk-go-v2/config"
+	"github.com/aws/aws-sdk-go-v2/feature/rds/auth"
+	"github.com/go-sql-driver/mysql"
+	"go.uber.org/zap"
+)
+
+type client interface {
+	Connect() error
+	getRecords(dbquery *DBQueries) (map[string]string, error)
+	Close() error
+}
+
+type mySQLClient struct {
+	connStr string
+	client  *sql.DB
+	logger  *zap.Logger
+	conf    *Config
+}
+
+var _ client = (*mySQLClient)(nil)
+
+//This function is used for reading certificates from .pem file for different AWS regions and passing them on as a tls config for authentication.
+//Details : https://docs.aws.amazon.com/AmazonRDS/latest/UserGuide/UsingWithRDS.SSL.html
+func createIAMRDSTLSConf(pempath string, logger *zap.Logger) tls.Config {
+	rootCertPool := x509.NewCertPool()
+	globalpem, err := ioutil.ReadFile(pempath)
+	if err != nil {
+		logger.Error("error in reading pem file", zap.Error(err))
+	}
+	if ok := rootCertPool.AppendCertsFromPEM(globalpem); !ok {
+		logger.Error("error in loading certificates from pem file", zap.Error(err))
+	}
+	return tls.Config{
+		RootCAs: rootCertPool,
+	}
+}
+
+//This function calls for the AWS packaged API which will generate an authentication token that can be used for accessing a AWS RDS instance instead of a password.
+//Details : https://docs.aws.amazon.com/AmazonRDS/latest/UserGuide/UsingWithRDS.IAMDBAuth.html
+func generateIAMAuthToken(endpoint string, conf *Config, logger *zap.Logger) (token string) {
+	cfg, err := config.LoadDefaultConfig(context.TODO())
+	if err != nil {
+		logger.Error("configuration error:", zap.Error(err))
+	}
+	authenticationToken, err := auth.BuildAuthToken(
+		context.TODO(), endpoint, conf.Region, conf.Username, cfg.Credentials)
+	if err != nil {
+		logger.Error("failed to create authentication token:", zap.Error(err))
+	}
+	return authenticationToken
+}
+
+//There are 3 scenarios here for creating connection strings for a database connection
+//1. With a plaintext password
+//2. With an encrypted plaintext password
+//3. With an AWS Authentication token to be used as a password
+func newMySQLClient(conf *Config, logger *zap.Logger) client {
+	var port string
+	var basicauthpassword string
+	var connStr string
+	basicauthpassword = conf.Password
+	//Encrypting a plaintext password if a 24 character secret string is provided by the user from an external file
+	if (len(conf.PasswordType) == 0 || conf.PasswordType == "plaintext") && len(conf.EncryptSecretPath) != 0 {
+		MySecret1, err := readMySecret(conf)
```

Comment:
```suggestion
		secret, err := readMySecret(conf)
```

I don't see a reason to use `My` prefix

also variables in functions should starting with lower case

Reply:
adding number to variable is not so informative

Reply:
will modify this.